### ISSUE TITLE
shebang interpreter cannot open it's file

### ISSUE THREAD

**Issue by inclyc** [Mic92/envfs#114]
Recently I tried to open "home-manager" in "envfs". However, looks like we correctly invokes "bash" interpreter, but it cannot find "/usr/bin/home-manager" source file in `/usr/bin/`.

```
$ /usr/bin/home-manager
/nix/store/j80wcbm6bb1nsqsgmniq43rg980p27q9-bash-5.2-p15/bin/bash: /usr/bin/home-manager: No such file or directory
```

Hmm, recently I made duplicated work on my own repo before finding out this one :(  based on a different approach: libfuse3 which does not seem to have this issue.

https://github.com/inclyc/pathfs

Edit: related to "stat" operation?

**Comment by SuperSandro2000** [comment]
I encountered the same problem with anything that is a shellscript.

**Comment by Mic92** [comment]
Supporting stat is not feasible I think. I had a couple of instances where programs checked for an executable with stat and start the program at this path without $PATH set. I rather have some instances not working than breaking podman and GNOME.

**Comment by Mic92** [comment]
You should not have the mountpoint of envfs in your $PATH as this would lead to these type of breakages.

**Comment by Mic92** [comment]
Maybe 'stat()' could be allowed if a program tries to query itself?

**Comment by SuperSandro2000** [comment]
I think the minimal reproducer could already be trying to run a shellscript with a patched shebang but I didn't find the time to verify this.

**Comment by Mic92** [comment]
Still an issue with https://github.com/Mic92/envfs/pull/146 ?

**Comment by SuperSandro2000** [comment]
I think this is fixed on 1.0.5 but I only did basic testing with a basic script.
